operator/v1/types_network.go

```diff
@@ -788,3 +814,38 @@ const (
 	// between pods on the cluster network.
 	IPsecModeFull IPsecMode = "Full"
 )
+
+// +kubebuilder:validation:Enum:="";"Enabled";"Disabled"
+type RouteAdvertisementsEnablement string
+
+var (
+	// RouteAdvertisementsEnabled enables route advertisements for ovn-kubernetes
+	RouteAdvertisementsEnabled RouteAdvertisementsEnablement = "Enabled"
+	// RouteAdvertisementsDisabled disables route advertisements for ovn-kubernetes
+	RouteAdvertisementsDisabled RouteAdvertisementsEnablement = "Disabled"
+)
+
+// RoutingCapabilitiesProvider is a component providing routing capabilities.
+// +kubebuilder:validation:Enum=FRR-K8s
+type RoutingCapabilitiesProvider string
+
+const (
+	// RoutingCapabilitiesProviderFRRK8s determines FRR-K8s is providing
+	// advanced routing capabilities.
+	RoutingCapabilitiesProviderFRRK8s RoutingCapabilitiesProvider = "FRR-K8s"
```

Comment:
AdvancedRoutingProviderFRR AdvancedRoutingProvider = "FRR" ?

I think FRR-K8S can just be implied when you enable FRR.

Reply:
I touched on this on a previous comment as well.
https://github.com/openshift/api/pull/1944#discussion_r1683143806

The reason to go for "FFR-K8S" is that "FFR-K8S" might not be the only project deploying FRR. In the future, a different project may come around that deploys FRR as well and for whatever reason we might want to make it available too as an alternative, addition or replacement with a transition period. Even if unlikely,  referring to what we are actually deploying gets us out of that trouble.

Reply:
FRR-K8S is not "deploying FRR". FRR is going to be deployed by CNO. FRR-K8S is another thing installed by CNO to do configuration of  FRR via KAPI. I think you are suggesting that down the road the configuration software changes from FRR-K8S to FRR-NEW-K8S, and cant update the value here or something?

The core technology is here (that is actually doing the routing function) is FRR, so that's why I feel that it is better to use that.

Reply:
Prefer to name things based on the user facing feature over the implementation details, so, I'm with Tim on this one

Reply:
I don't see it this way. CNO is not deploying FRR. CNO is **just** deploying "FRR-K8S" and "FRR" is a daemon that lives in one of the "FRR-K8S" containers: https://github.com/openshift/frr/blob/377d02f94181d26dac3ab2e2067ced9fe8d73763/Dockerfile.openshift#L42

Additionally, the user needs to be aware that "FRR-K8S" is being deployed as the API the user will be interacting with is specific to "FRR-K8S" (even if logically it can be mapped to an FRR configuration).

Finally, there can exist another project X that also "contains" FRR but with a very different configuration API.

Anyway, I have tried to explain this many times here and been unsuccessful. So I will take that as a sign that it is just better/easier for everyone for this to be understood just as "FRR" so will change accordingly.